app/support/order_row_importer.rb

```diff
@@ -70,10 +70,14 @@ def order_key
     @order_key ||= [user_field, chart_string_field, order_date_field]
   end
 
-  def row_with_errors # TODO: refactor
-    new_row = @row.dup
+  def row_with_errors
+    # Start with a hash of HEADERS keys with nil values to ensure optional columns
+    # are included in the report even if they are not in the uploaded CSV.
```

Comment:
Right. Okay, I see that now. Might be a clearer way to create that empty hash. Would this work?

```
Headers.values.each_with_object({}) { |value, hash| hash[value] = nil }
```